### PR TITLE
Adding third party module - example

### PR DESCRIPTION
**Pull request by augustocollado**
### Description

This PR completes the third-party module article by illustrating the process of adding a third-party pallet, the error you'll get when trying to compile, and how to apply the solution by patching the deps with dinner.
This section relies on this other repo containing the pallet: https://github.com/papermoonio/pallet-toggle

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] If this page requires a disclaimer, I have added one
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/build/customize/adding-external-module.md [hunk 1]
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+pallet-toggle = { 
+    git = "https://github.com/papermoonio/pallet-toggle.git", 
+    default-features = false 
+}
+...
+```
+
+#### Make the Standard Features Available to the Compiler {: #add-standard-features }
+
+Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which is located in the folder `container-chains/templates/simple/runtime`.
```
Comment (albertov19) [inline builders/build/customize/adding-external-module.md:65]:
```suggestion
Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which for this example that uses the Tanssi GitHub repo, is located in the folder `container-chains/templates/simple/runtime`.
```

File: builders/build/customize/adding-external-module.md [hunk 2]
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+	"pallet-toggle/runtime-benchmarks",
+]
+
+try-runtime = [
+	...,
+	"pallet-toggle/try-runtime",
+]
+```
+
+#### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
```
Comment (albertov19) [inline builders/build/customize/adding-external-module.md:98]:
```suggestion
### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
```

File: builders/build/customize/adding-external-module.md [hunk 3]
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+    pub enum Runtime
+    {
+        ...
+        ...
+        Toggle: pallet_toggle,
+    }
+);
+```
+
+### Compile {: #compile }
```
Comment (albertov19) [inline builders/build/customize/adding-external-module.md:119]:
```suggestion
### Compile Runtime {: #compile-runtime }
```

File: builders/build/customize/adding-external-module.md [hunk 4]
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+
+try-runtime = [
+	...,
+	"pallet-toggle/try-runtime",
+]
+```
+
+#### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
+
+Add the following snippet that configures the module and add the module within the `construct_runtime!`macro.
```
Comment (albertov19) [inline builders/build/customize/adding-external-module.md:100]:
```suggestion
Next, add the following snippet in the `lib.rs` file inside the runtime folder. This configures the module and add the module within the `construct_runtime!`macro.
```

File: builders/build/customize/adding-external-module.md [hunk 5]
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+
+The terminal output will display an error, similar to the following, caused by different modules referencing different versions of the same dependency:
+
+```bash
+error: failed to select a version for `syn`.
+```
+
+### Patch depedencies {: #patch-dependencies }
+
+Finally, executing the `diener` [command](#solving-dependencies-conflicts-diener) will add a `patch` section to your workspace `Cargo.toml` file, overriding the dependencies and unifying origins and versions.
```
Comment (albertov19) [inline builders/build/customize/adding-external-module.md:137]: Add an example of how the diener command is executed for this example

**Review by albertov19 — CHANGES_REQUESTED** [review]
Left some feedback to improve readability :)

**Review by albertov19 — CHANGES_REQUESTED** [review]
Looks great! As my last piece of feedback, I would use now `termynal` 

It will be an easy thing to adapt here. Check the reference PR -> https://github.com/moondance-labs/tanssi-docs/pull/74

File: builders/build/customize/adding-external-module.md [hunk 6]
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
 This command applies the changes to the `Cargo.toml` file, patching the dependencies, and solving the double reference issues.
 
 You can visit the [diener documentation](https://docs.rs/crate/diener/latest){target=_blank} to learn more about the tool and other extra functions it offers.
+
+## Example of the Double Reference Issue {: #double-reference-issue }
+
+To illustrate the situation, the following steps add a demo [external module](https://github.com/papermoonio/pallet-toggle.git){target=_blank} to a custom runtime based on the [baseline Appchain template](/builders/build/templates/substrate/){target=_blank}. One way to follow this tutorial is to clone the [Tanssi Github repository](https://github.com/moondance-labs/tanssi){target=_blank}, which will act as the root repository of the project.
```
Comment (eshaben) [inline builders/build/customize/adding-external-module.md:35]: All of the `target=_blank`s on this page should be changed to `target=\_blank`

File: builders/build/customize/adding-external-module.md [hunk 7]
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+
+To resolve this issue, it will be necessary to apply a patch so that the references for the dependency are unified:
+
+![Patched reference](/images/builders/build/external-module/external-module-2.png)
+
+### Declaring the Dependency {: #declaring-dependency }
+
+The first step to reproduce the double reference issue is to declare the dependency in the `Cargo.toml` file located in the repository's root folder, under the section `[dependencies]`. For this example, a simple [toggle module](https://github.com/papermoonio/pallet-toggle.git){target=_blank} is used. 
+
+This `toggle` module, built for testing and educational purposes, adds a basic logic to the runtime, allowing users to switch a state between true and false.
```
Comment (eshaben) [inline builders/build/customize/adding-external-module.md:61]:
```suggestion
This `toggle` module, built for testing and educational purposes, adds basic logic to the runtime, allowing users to switch a state between true and false.
```

File: builders/build/customize/adding-external-module.md [hunk 8]
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+pallet-toggle = { 
+    git = "https://github.com/papermoonio/pallet-toggle.git", 
+    default-features = false 
+}
+...
+```
+
+### Make the Standard Features Available to the Compiler {: #add-standard-features }
+
+Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which for this example that uses the Tanssi GitHub repo, is located in the folder `container-chains/templates/simple/runtime`.
```
Comment (eshaben) [inline builders/build/customize/adding-external-module.md:75]:
```suggestion
Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which, for this example that uses the Tanssi GitHub repo, is located in the folder `container-chains/templates/simple/runtime`.
```

File: builders/build/customize/adding-external-module.md [hunk 9]
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+
+try-runtime = [
+	...,
+	"pallet-toggle/try-runtime",
+]
+```
+
+### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
+
+Next, add the following snippet in the `lib.rs` file inside the runtime folder. This configures the module and add the module within the `construct_runtime!`macro.
```
Comment (eshaben) [inline builders/build/customize/adding-external-module.md:110]:
```suggestion
Next, add the following snippet to the `lib.rs` file inside the runtime folder. This configures the module and adds the module within the `construct_runtime!` macro.
```

File: builders/build/customize/adding-external-module.md [hunk 10]
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
...
+cargo build -p container-chain-template-simple-node --release
+```
+
+The terminal output will display an error, similar to the following, caused by different modules referencing different versions of the same dependency:
+
+```bash
+error: failed to select a version for `syn`.
+```
+
+### Patch depedencies {: #patch-dependencies }
```
Comment (eshaben) [inline builders/build/customize/adding-external-module.md:145]:
```suggestion
### Patch Dependencies {: #patch-dependencies }
```

**Comment by albertov19** [comment]
@augustocollado you can remove the terminal image then :)

**Comment by augustocollado** [comment]
> @augustocollado you can remove the terminal image then :)

Ouch! I deleted the image, but I don't know how that copy with that name got in that new folder I created. Fixed.

> @augustocollado don't forget the `target=\_blank` as per the changes made in this PR: #76

Sorry about that, I thought I had gotten them all :S

**Review by eshaben — APPROVED** [review]
LGTM!

**Review by albertov19 — APPROVED** [review]
LGTM

**Pull request merged**